moit-api/src/main/kotlin/com/mashup/moit/infra/fcm/FCMNotificationService.kt

```diff
@@ -3,37 +3,56 @@ package com.mashup.moit.infra.fcm
 import com.google.firebase.messaging.FirebaseMessaging
 import com.google.firebase.messaging.Message
 import com.google.firebase.messaging.Notification
+import com.mashup.moit.common.exception.MoitException
+import com.mashup.moit.common.exception.MoitExceptionType
 import com.mashup.moit.domain.moit.NotificationRemindOption
+import com.mashup.moit.domain.notification.UrlSchemeProperties
+import com.mashup.moit.domain.notification.generator.ScheduledStudyNotificationGenerator
+import com.mashup.moit.domain.notification.generator.StartAttendanceNotificationGenerator
 import org.slf4j.LoggerFactory
 import org.springframework.stereotype.Service
 
 @Service
 class FCMNotificationService(
-    private val firebaseMessaging: FirebaseMessaging
+    private val firebaseMessaging: FirebaseMessaging,
+    private val urlSchemeProperties: UrlSchemeProperties
 ) {
     private val logger = LoggerFactory.getLogger(FCMNotificationService::class.java)
 
-    fun pushStartStudyNotification(startNotification: StudyAttendanceStartNotification) {
-        val topic = getMoitTopic(startNotification.moitId)
-        try {
+    fun pushStudyNotification(studyNotification: StudyNotification) {
+        val topic = getMoitTopic(studyNotification.moitId)
+
+        runCatching {
             val notification = Notification.builder()
-                .setTitle(startNotification.title)
-                .setBody(startNotification.body)
+                .setTitle(studyNotification.title)
+                .setBody(studyNotification.body)
                 .build()
 
+            val data = when (studyNotification) {
+                is ScheduledStudyNotification ->
+                    mapOf("urlScheme" to ScheduledStudyNotificationGenerator.urlScheme(urlSchemeProperties.studyScheduled, studyNotification.moitId))
+
+                is StudyAttendanceStartNotification ->
+                    mapOf("urlScheme" to StartAttendanceNotificationGenerator.urlScheme(urlSchemeProperties.attendanceStart, studyNotification.moitId))
+
+                else -> throw MoitException.of(MoitExceptionType.SYSTEM_FAIL)
+            }
```

Comment:
Scheme 선택하는 부분이랑 data를 구성하는 부분 나눠봐도 좋을 것 같습니다.
```suggestion
            val urlScheme = when (studyNotification) {
                is ScheduledStudyNotification -> urlSchemeProperties.studyScheduled
                is StudyAttendanceStartNotification -> urlSchemeProperties.attendanceStart
                else -> throw MoitException.of(MoitExceptionType.SYSTEM_FAIL)
            }
            val data = mapOf("urlScheme" to ScheduledStudyNotificationGenerator.urlScheme(urlScheme, studyNotification.moitId))
```

위 UrlScheme 구하는 부분을 아예 메서드로 빼면 더 깔끔해질 것 같기두 하구요 ㅎㅎ
필수는 아니고 그냥 의견입니다 :)

Reply:
중복 코드는 줄이는 것이 좋죠 적용하였습니다 감사합니다!